### DISCUSSION TITLE
Where can I run Android-specific code before OnFrameworkInitializationCompleted?

### DISCUSSION THREAD

**Discussion by tristanlabelle** [AvaloniaUI/Avalonia#9398] [Q&A] [answered]
I would like to have a chance to register Android-specific services for dependency injection before `OnFrameworkInitializationCompleted`, since I need to obtain them when building my UI. I noticed that `OnFrameworkInitializationCompleted` is called before my `MainActivity` object is constructed. The only other place I can think of is the `SplashActivity`, but is it guaranteed to run before `OnFrameworkInitializationCompleted`? Where else could I put it?

**Comment by maxkatz6** [comment, marked as answer]
Avalonia Application is created inside of [AvaloniaSplashActivity.OnCreate](https://github.com/AvaloniaUI/Avalonia/blob/master/src/Android/Avalonia.Android/AvaloniaSplashActivity.cs#L11). So, you can expect that OnFrameworkInitializationCompleted is executed right after that (after SetupWithLifetime was called from the splash activity).

**Reply by tristanlabelle** [reply]
That worked, thanks!
